src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletService.java

```diff
@@ -7,4 +7,6 @@
 
 public interface WalletService {
     WalletDTO createWallet(@Valid CreateWalletDTO createWalletDTO);
+
+    WalletDTO findById(long id);
```

Comment:
Please change it to:
```java
 WalletDTO findById(@Min(value = 1, message = "Wallet id has to be greater than 0") Long id)
```

The primitive variable 'long' cannot be 'null' value here.
Long can - and if he does, validation will catch him. So it's safer to catch the appropriate validation exception and process it as we want.